pyaerocom/vert_coords.py

```diff
@@ -54,14 +57,17 @@ def atmosphere_sigma_coordinate_to_pressure(sigma, ps, ptop):
         computed pressure levels in Pa (standard_name=air_pressure)
     """
     if not isinstance(ptop, float):
-        try:
-            ptop = float(ptop)
-        except Exception as e:
-            raise ValueError(f"Invalid input for ptop. Need floating point\nError: {repr(e)}")
+        if not isinstance(ptop, np.ndarray):
+            try:
+                ptop = float(ptop)
+            except ValueError as e:
+                raise ValueError(f"Invalid input for ptop. Need floating point\nError: {repr(e)}")
```

Comment:
you can simplify the error section with 
``` python
    if not isinstance(ptop, float | np.ndarray):
        try:
            ptop = float(ptop)
        except ValueError as e:
            raise ValueError("Invalid input for ptop. Need floating point") from e
```

Using `|` on `isinstance(ptop, float | np.ndarray)` is a [Python 3.10 feature](https://docs.python.org/3/library/stdtypes.html#types-union).

The mechanism behind `raise ValueError(...) from e` is called [exception chaining](https://docs.python.org/3/tutorial/errors.html#exception-chaining), and will show the original exception as part of the stack trace.

Reply:
Fixed.